diff -urN a/nbtscan.c b/nbtscan.c
--- a/nbtscan.c	2026-03-09 18:40:06.686357067 +0100
+++ b/nbtscan.c	2026-03-09 18:40:28.163023731 +0100
@@ -131,24 +131,24 @@
     printf("Version major: 0x%02x (%1$d)\n", hostinfo->footer->version_major);
     printf("Version minor: 0x%02x (%1$d)\n", hostinfo->footer->version_minor);
     printf("Duration: 0x%04x (%1$d)\n", hostinfo->footer->duration);
-    printf("FRMRs Received: 0x%04 (%1$d)\n", hostinfo->footer->frmps_received);
-    printf("FRMRs Transmitted: 0x%04 (%1$d)\n", hostinfo->footer->frmps_transmitted);
-    printf("IFrame Receive errors: 0x%04 (%1$d)\n", hostinfo->footer->iframe_receive_errors);
-    printf("Transmit aborts: 0x%04 (%1$d)\n", hostinfo->footer->transmit_aborts);
-    printf("Transmitted: 0x%08 (%1$d)\n", hostinfo->footer->transmitted);
-    printf("Received: 0x%08 (%1$d)\n", hostinfo->footer->received);
-    printf("IFrame transmit errors: 0x%04 (%1$d)\n", hostinfo->footer->iframe_transmit_errors);
-    printf("No receive buffers: 0x%04 (%1$d)\n", hostinfo->footer->no_receive_buffer);
-    printf("tl timeouts: 0x%04 (%1$d)\n", hostinfo->footer->tl_timeouts);
-    printf("ti timeouts: 0x%04 (%1$d)\n", hostinfo->footer->ti_timeouts);
-    printf("Free NCBS: 0x%04 (%1$d)\n", hostinfo->footer->free_ncbs);        
-    printf("NCBS: 0x%04 (%1$d)\n", hostinfo->footer->ncbs);
-    printf("Max NCBS: 0x%04 (%1$d)\n", hostinfo->footer->max_ncbs);
-    printf("No transmit buffers: 0x%04 (%1$d)\n", hostinfo->footer->no_transmit_buffers);
-    printf("Max datagram: 0x%04 (%1$d)\n", hostinfo->footer->max_datagram);
-    printf("Pending sessions: 0x%04 (%1$d)\n", hostinfo->footer->pending_sessions);
-    printf("Max sessions: 0x%04 (%1$d)\n", hostinfo->footer->max_sessions);
-    printf("Packet sessions: 0x%04 (%1$d)\n", hostinfo->footer->packet_sessions);
+    printf("FRMRs Received: 0x%04x (%1$d)\n", hostinfo->footer->frmps_received);
+    printf("FRMRs Transmitted: 0x%04x (%1$d)\n", hostinfo->footer->frmps_transmitted);
+    printf("IFrame Receive errors: 0x%04x (%1$d)\n", hostinfo->footer->iframe_receive_errors);
+    printf("Transmit aborts: 0x%04x (%1$d)\n", hostinfo->footer->transmit_aborts);
+    printf("Transmitted: 0x%08x (%1$d)\n", hostinfo->footer->transmitted);
+    printf("Received: 0x%08x (%1$d)\n", hostinfo->footer->received);
+    printf("IFrame transmit errors: 0x%04x (%1$d)\n", hostinfo->footer->iframe_transmit_errors);
+    printf("No receive buffers: 0x%04x (%1$d)\n", hostinfo->footer->no_receive_buffer);
+    printf("tl timeouts: 0x%04x (%1$d)\n", hostinfo->footer->tl_timeouts);
+    printf("ti timeouts: 0x%04x (%1$d)\n", hostinfo->footer->ti_timeouts);
+    printf("Free NCBS: 0x%04x (%1$d)\n", hostinfo->footer->free_ncbs);        
+    printf("NCBS: 0x%04x (%1$d)\n", hostinfo->footer->ncbs);
+    printf("Max NCBS: 0x%04x (%1$d)\n", hostinfo->footer->max_ncbs);
+    printf("No transmit buffers: 0x%04x (%1$d)\n", hostinfo->footer->no_transmit_buffers);
+    printf("Max datagram: 0x%04x (%1$d)\n", hostinfo->footer->max_datagram);
+    printf("Pending sessions: 0x%04x (%1$d)\n", hostinfo->footer->pending_sessions);
+    printf("Max sessions: 0x%04x (%1$d)\n", hostinfo->footer->max_sessions);
+    printf("Packet sessions: 0x%04x (%1$d)\n", hostinfo->footer->packet_sessions);
   };
 };
 
